changedetectionio/fetch_site_status.py

```diff
@@ -104,9 +104,16 @@ def run(self, uuid):
             # https://stackoverflow.com/questions/41817578/basic-method-chaining ?
             # return content().textfilter().jsonextract().checksumcompare() ?
 
-            is_html = True
+            is_json = fetcher.headers.get('Content-Type', '') == 'application/json'
+            is_html = not is_json
             css_filter_rule = watch['css_filter']
-            if css_filter_rule and len(css_filter_rule.strip()):
+
+            has_filter_rule = css_filter_rule and len(css_filter_rule.strip())
+            if is_json and not has_filter_rule:
+                css_filter_rule = "json:$"
```

Comment:
This rule should match the whole json response, shouldn't it?

Reply:
Ah man, i've been staring at that code for 10 minutes and I can't understand it .. needs a serious refactor